pkg/controllers/securitypolicy_controller.go

```diff
@@ -260,33 +273,45 @@ func (r *SecurityPolicyReconciler) GarbageCollector(cancel chan bool, timeout ti
 			return
 		case <-time.After(timeout):
 		}
-		nsxPolicySet := r.Service.ListSecurityPolicyID()
-		if len(nsxPolicySet) == 0 {
+
+		keys := r.Service.ListSecurityPolicyKeys()
```

Comment:
Could you help explain why you update logic in GarbageCollector()? I did not find anything different or better between existing and new implementation.